EventFilter/Utilities/interface/EvFOutputModule.h

```diff
@@ -0,0 +1,118 @@
+#ifndef EventFilter_Utilities_EvFOutputModule_h
+#define EventFilter_Utilities_EvFOutputModule_h
+
+#include "IOPool/Streamer/interface/StreamerOutputFile.h"
+#include "FWCore/Framework/interface/one/OutputModule.h"
+#include "IOPool/Streamer/interface/StreamerOutputModuleCommon.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+
+#include "EventFilter/Utilities/interface/JsonMonitorable.h"
+#include "EventFilter/Utilities/interface/FastMonitor.h"
+#
+typedef edm::detail::TriggerResultsBasedEventSelector::handle_t Trig;
+
+namespace evf {
+
+  class FastMonitoringService;
+//  class EvFOutputEventWriter;
+
+  class EvFOutputEventWriter
+  {
+  public:
+
+    explicit EvFOutputEventWriter(std::string const& filePath):
+      filePath_(filePath),
+      accepted_(0),
+      stream_writer_events_(new StreamerOutputFile(filePath))
+    {
+    }
+
+    ~EvFOutputEventWriter() {
+    }
+
+    void close() {
+        stream_writer_events_->close();
+    }
+
+    void doOutputEvent(EventMsgBuilder const& msg) {
+      EventMsgView eview(msg.startAddress());
+      stream_writer_events_->write(eview);
+    }
+
+    uint32 get_adler32() const {
+      return stream_writer_events_->adler32();
+    }
+
+    std::string const& getFilePath() const {return filePath_;}
+
+    unsigned long getAccepted() const {return accepted_;}
+    void incAccepted() {accepted_++;}
+
+  private:
+    std::string filePath_;
+    unsigned long accepted_;
+    edm::propagate_const<std::shared_ptr<StreamerOutputFile>> stream_writer_events_;
```

Comment:
Wouldn't std::unique_ptr be more appropriate as only this class is allowed to own the object?

Reply:
I can agree it would be more appropriate.
By looking more closely at this, there is another usage of shared_ptr, "stream_writer_preamble" in beginRun. It looks like here just a local variable can be used, so I will push that change as well.